frontend/src/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect } from 'react';
+import { Breadcrumb, Layout, Menu } from 'antd';
+import styles from './Menu.module.css';
+import { items } from '../../consts/MenuItem';
+import { MenuItem } from '../../types';
+import { Outlet } from 'react-router-dom';
+
+const { Header, Content, Footer, Sider } = Layout;
+
+const LeftMenu: React.FC = () => {
+    const [collapsed, setCollapsed] = useState<boolean>(false);
+    const [currentPage, setCurrentPage] = useState<string>('');
+    const [selectedKey, setSelectedKey] = useState<string>('1');
+    const [asideWidth, setAsideWidth] = useState<number>(250);
+    const username = 'k8_userjjjjjjjjjjjjjjjiiiiiiiii';
+
+    const findItemByKey = (key: string, items: MenuItem[]): MenuItem | { sectionLabel: string, childLabel: string } | null => {
+        for (const item of items) {
+            if (item.key === key) {
+                return item;
+            }
+            // If the item has children, search recursively in children
+            if (item.children) {
+                const found = findItemByKey(key, item.children);
+                if (found) {
+                    return {
+                        sectionLabel: item.label,
+                        childLabel: (found as MenuItem).label,
+                    };
+                }
+            }
+        }
+        return null;
+    };
+
+    const setCurrentPageFromItem = (item: MenuItem | { sectionLabel: string, childLabel: string }) => {
+        if ('childLabel' in item) {
+            setCurrentPage(`${item.sectionLabel}/${item.childLabel}`);
+        } else {
+            setCurrentPage(item.label);
+        }
+    };
+
+    const handleMenuClick = (e: { key: string }) => {
+        setSelectedKey(e.key);
+        const selectedItem = findItemByKey(e.key, items);
+
+        if (selectedItem) {
+            setCurrentPageFromItem(selectedItem);
+        } else {
+            setCurrentPage('');
+        }
+    };
+
+    // Effect to set the default page title based on the initially selected key
+    useEffect(() => {
+        const defaultItem = findItemByKey(selectedKey, items);
+        if (defaultItem) {
+            setCurrentPageFromItem(defaultItem);
+        }
+    }, []);
+
+    return (
+        <Layout className={styles.menuLayout}>
+            <Sider
+                className={styles.menuSider}
+                collapsible
+                collapsed={collapsed}
+                onCollapse={(value) => {
+                    setCollapsed(value);
+                    setAsideWidth(asideWidth === 80 ? 250 : 80);
+                }}
+                width={`${asideWidth}px`}
+            >
+                <div className={styles.logo}>
+                    <span className={styles.logoText}>
+                        {collapsed ? 'U' : username.length > 10 ? `${username.slice(0, 10)}...` : username}
+                    </span>
+                </div>
```

Comment:
Ignore this for now, but if we have time in the future maybe add some tooltip on hover.

Reply:
I'll do that when I'll create a page with user information.